File: hyein/week10/N으로표현.py
def solution(N, number):
    dp  =[set([N*int('1'*i)]) for i in range(1, 9)] 

    for i in range(8): 
        for j in range(i):
            for num1 in dp[j]: 
                for num2 in dp[i-j-1]: 
                    dp[i].add(num1 + num2)
                    dp[i].add(num1 - num2)
                    dp[i].add(num1 * num2)
                    if num2 != 0:
                        dp[i].add(num1//num2)
                        
        if number in dp[i]:
            return i+1 
    return -1

File: hyejee/week10/N으로 표현.cpp
#include <string>
#include <unordered_set>
#include <vector>

using namespace std;

int solution(int N, int number) {
    int answer = -1; 
    unordered_set<int> s[8]; 
    
    int sum = 0;
    for(int i = 0; i < 8; ++i) { 
        sum = 10*sum + N;
        s[i].insert(sum);
    }
    
    for (int i = 1; i < 8; ++i) {
        for (int j = 0; j < i; ++j) {
            for (int a : s[j]) {
                for(int b : s[i-j-1]) {
                    s[i].insert(a+b);
                    s[i].insert(a-b);
                    s[i].insert(a*b);
                    if (b != 0)
                        s[i].insert(a/b);
                }
            }
        }
    }
    
    for (int i = 0; i < 8; ++i) {
        if (s[i].find(number) != s[i].end()) {
            answer = i + 1;
            break;
        }  
    }
        
    return answer;
}
